fleetshard/pkg/central/reconciler/reconciler.go

```diff
@@ -218,18 +223,23 @@ func (r *CentralReconciler) Reconcile(ctx context.Context, remoteCentral private
 	}
 
 	// TODO(create-ticket): When should we create failed conditions for the reconciler?
-	return r.readyStatus(ctx, remoteCentralNamespace)
-}
-
-func (r *CentralReconciler) readyStatus(ctx context.Context, namespace string) (*private.DataPlaneCentralStatus, error) {
 	status := readyStatus()
-	if r.useRoutes {
-		routesStatuses, err := r.getRoutesStatuses(ctx, namespace)
+	// Do not report routes statuses if:
+	// 1. Routes are not used on the cluster
+	// 2. Central request is in status "Ready" - assuming that routes are already reported and saved
+	if r.useRoutes && !remoteCentralReady {
+		status.Routes, err = r.getRoutesStatuses(ctx, remoteCentralNamespace)
 		if err != nil {
 			return nil, err
 		}
-		status.Routes = routesStatuses
 	}
+
+	// Cache MUST be updated right before returning "ready" result.
+	// Otherwise, the hash will be saved without sending the actual status (in case of error).
+	if err := r.setLastCentralHash(remoteCentral); err != nil {
+		return nil, errors.Wrapf(err, "setting central reconcilation cache")
```

Comment:
nit: adding the cluster ID?
```suggestion
		return nil, errors.Wrapf(err, "updating reconcilation cache for cluster %s", remoteCentral.Id)
```

Reply:
This is already done by caller (runtime): 
```
			glog.Errorf("Unexpected error occurred for central %s/%s: %v", central.Metadata.Namespace, central.Metadata.Name, err.Error())
```